.github/workflows/build-test.yml

```diff
@@ -14,129 +14,578 @@ on:
         cat /etc/os-release
-    - name: Initialize git submodules
+    -
+      name: Set up QEMU
+      id: qemu
+      uses: docker/setup-qemu-action@v1
+      with:
+        image: tonistiigi/binfmt:latest
+        platforms: linux/arm64
+    -
+      name: Available platforms
+      run: echo ${{ steps.qemu.outputs.platforms }}
+    -
+      name: Initialize git submodules
+      run: |
+        git submodule update --init --recursive
+    -
+      name: Build RedBPF with the kernel headers on aarch64 Ubuntu 20.04 LTS
+      run: |
+        docker run --privileged \
+        -v $PWD:/build \
+        -w /build \
+        $REDBPF_IMAGE_NAME:latest-aarch64-ubuntu20.04 \
+        /bin/bash -c 'export KERNEL_SOURCE=$(echo /lib/modules/*) \
+          && echo KERNEL_SOURCE=$KERNEL_SOURCE \
+          && cargo clean \
+          && cargo build --no-default-features --features=llvm13 \
+          && cargo build --no-default-features --features=llvm13,command-line --bin cargo-bpf \
+          && cargo build --no-default-features --features=llvm13 --examples'
+
+  build-on-aarch64-ubuntu-1804:
+    runs-on: ubuntu-latest
+    steps:
+    -
+      name: Checkout
+      uses: actions/checkout@v2
+    -
+      name: Build host info
+      run: |
+        uname -a
+        cat /etc/os-release
+    -
+      name: Set up QEMU
+      id: qemu
+      uses: docker/setup-qemu-action@v1
+      with:
+        image: tonistiigi/binfmt:latest
+        platforms: linux/arm64
+    -
+      name: Available platforms
+      run: echo ${{ steps.qemu.outputs.platforms }}
+    -
+      name: Initialize git submodules
       run: |
         git submodule update --init --recursive
-    - name: Run RedBPF build on alpine-3.14 container
+    -
+      name: Build RedBPF with the kernel headers on aarch64 Ubuntu 18.04 LTS
       run: |
         docker run --privileged \
         -v $PWD:/build \
         -w /build \
-        $BASE_IMAGE:latest-alpine \
-        sh -c 'cargo clean && export KERNEL_VERSION=$(ls --indicator-style=none /lib/modules/) RUSTFLAGS=-Ctarget-feature=-crt-static && echo KERNEL_VERSION=$KERNEL_VERSION && cargo +1.51 build --no-default-features --features llvm11 && cargo +1.51 build --no-default-features --features llvm11 --examples'
+        $REDBPF_IMAGE_NAME:latest-aarch64-ubuntu18.04 \
+        /bin/bash -c 'export KERNEL_SOURCE=$(echo /lib/modules/*) \
+          && echo KERNEL_SOURCE=$KERNEL_SOURCE \
+          && cargo clean \
+          && cargo build --no-default-features --features=llvm13 \
+          && cargo build --no-default-features --features=llvm13,command-line --bin cargo-bpf \
+          && cargo build --no-default-features --features=llvm13 --examples'
 
   publish:
     runs-on: ubuntu-latest
     if: startsWith( github.ref, 'refs/tags/v')
     needs:
-    - fedora-34-build-test
-    - ubuntu-2104-build-test
-    - ubuntu-2004-build-test
-    - ubuntu-1804-build-test
-    - alpine-314-build-test
+    - build-on-x86_64-ubuntu-2104
```

Comment:
I think we should split this list into 2 targets, x64 and aarch64, so that they can run in parallel. Then we can also make any other higher-level target depend on these two.